src/Commands.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Commands.
+ */
+
+namespace P4\MasterTheme;
+
+use CF\Integration\DefaultConfig;
+use CF\Integration\DefaultIntegration;
+use CF\Integration\DefaultLogger;
+use CF\WordPress\DataStore;
+use CF\WordPress\WordPressAPI;
+use CF\WordPress\WordPressClientAPI;
+use WP_CLI;
+
+/**
+ * Class with a static function just because PHP can't autoload functions.
+ */
+class Commands {
+	/**
+	 * Add some WP_CLI commands if we're in CLI.
+	 */
+	public static function load() {
+		if ( ! defined( 'WP_CLI' ) || ! WP_CLI ) {
+			return;
+		}
+		$activator_command = static function ( $args, $assoc_args ) {
+			Activator::run();
+		};
+		WP_CLI::add_command( 'p4-run-activator', $activator_command );
+
+		/**
+		 * Put the CF API key into the options table, where the CF plugin uses it from.
+		 */
+		$put_cf_key_in_db = static function ( $args ) {
+			$hostname = $args[0];
+			if ( empty( $hostname ) ) {
+				WP_CLI::error( 'Please specify the hostname.' );
+			}
+
+			if ( ! defined( 'CLOUDFLARE_API_KEY' ) || empty( CLOUDFLARE_API_KEY ) ) {
+				WP_CLI::error( 'CLOUDFLARE_API_KEY constant is not set.' );
+			}
+
+			$domain_parts = explode( '.', $hostname );
+
+			$root_domain = implode( '.', array_slice( $domain_parts, - 2 ) );
+			update_option( 'cloudflare_api_key', CLOUDFLARE_API_KEY );
+			update_option( 'automatic_platform_optimization', [ 'value' => 1 ] );
+			update_option( 'cloudflare_cached_domain_name', $root_domain );
+		};
+		WP_CLI::add_command( 'p4-cf-key-in-db', $put_cf_key_in_db );
+
+		$purge_urls = static function ( $args, $assoc_args ) {
+			if ( ! Features::is_active( Features::CLOUDFLARE_DEPLOY_PURGE ) ) {
+				WP_CLI::warning( 'Purge on deploy is not enabled, not purging.' );
+
+				return;
+			}
+
+			if ( ! defined( 'CLOUDFLARE_PLUGIN_DIR' ) ) {
+				define( 'CLOUDFLARE_PLUGIN_DIR', WP_PLUGIN_DIR . '/cloudflare/' );
+			}
+			require_once CLOUDFLARE_PLUGIN_DIR . 'vendor/autoload.php';
```

Comment:
regular autoload doesn't work for this ? (it works locally without it at least)

Reply:
Not sure if it's really "autoloading" it, could be the classes happen to be loaded already at the point of execution. But if we don't add this then there's no explicit way of saying where these classes are coming from. If the plugin is not present this should also make it fail as early as possible.